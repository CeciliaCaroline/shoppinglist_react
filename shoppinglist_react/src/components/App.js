import React, {Component} from 'react';
import Register from './register';
import Login from "./login";
import Home from "./home";
import {BrowserRouter, Route} from 'react-router-dom';


class App extends Component {
    render() {
        return (
            <BrowserRouter>
                <div className="App">
                    <Route  exact path="/" component={Register}/>
                    <Route path="/login" component={Login}/>
                    <Route path="/home" component={Home}/>
                    {/*<Register/>*/}
                    {/*<Login/>*/}
                    {/*<Home/>*/}
                </div>
            </BrowserRouter>
        );
    }
}

export default App;
