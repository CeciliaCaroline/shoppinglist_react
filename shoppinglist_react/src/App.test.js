import React from 'react';
import ReactDOM from 'react-dom';
import { render, shallow} from 'enzyme';
import expect from 'expect';
import App from "./components/App";
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import "./setup_test";


// configure({adapter: new Adapter()});


describe('App component', () => {
    it('it renders app component', () => {

        const wrapper = render(<BrowserRouter
        ><App/>
        </BrowserRouter>);
        expect(wrapper.find('div').length).toEqual(8);

    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<App/>, div);
    });

    it('shallow renders without crashing', () => {
        shallow(<App/>);
    });

    it('renders correctly', () => {
        const tree = renderer.create(<App/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

