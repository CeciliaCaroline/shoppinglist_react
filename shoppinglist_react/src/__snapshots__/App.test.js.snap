// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="container mt-5"
  >
    <div
      className="row"
    >
      <div
        className="notifications-wrapper"
        style={Object {}}
      />
      <div
        className="container form-background card mx-5 mt-5 col media card "
      >
        <h2
          className="mt-5 text-capitalize"
        >
           Shopping list
        </h2>
        <h6
          className="text-capitalize"
        >
          The Best Way To Record and Share your shopping lists
        </h6>
        <p
          className="lead"
        >
          This is an application that enables users to record, track and share things they want to spend money on in order to meet their desires.
        </p>
        <h4>
          Create
        </h4>
        <p>
          Create a shopping list with all the items that you want to purchase
        </p>
        <h4>
          Update
        </h4>
        <p>
          Change your mind? Update your list with new items
        </p>
        <h4>
          Share
        </h4>
        <p>
          Share your list with friends and family
        </p>
      </div>
      <form
        className="container media-body form-background card mx-5 mt-5 col"
        onSubmit={[Function]}
      >
        <h2
          className="text-capitalize card-title mt-5"
        >
          New to Shoppinglist. Register
        </h2>
        <div
          className="form-group"
        >
          <label
            htmlFor="username"
          >
            Username
          </label>
          <input
            className="form-control"
            name="username"
            onChange={[Function]}
            required={true}
            type="text"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email
          </label>
          <input
            className="form-control"
            name="email"
            onChange={[Function]}
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="form-control"
            name="password"
            onChange={[Function]}
            required={true}
            type="password"
            value=""
          />
        </div>
        <div>
          <button
            className="btn btn-primary btn-block"
            onClick={[Function]}
            type="submit"
          >
            Sign up
          </button>
          <p
            className="text-center"
          >
            Already have an account??
            <a
              href="/auth/login"
              onClick={[Function]}
            >
              Login
            </a>
          </p>
        </div>
      </form>
    </div>
  </div>
</div>
`;
